akvo/rsr/static/scripts-src/project-directory.js

```diff
@@ -3,190 +3,148 @@
-      source: organisations.ttAdapter(),
-      templates: {
-        header: '<h3 class="dd-category">' + organisations_text + '</h3>',
-        suggestion: _.template('<a href="/projects/?organisation=<%= id %>"><p><%= name %></p></a>')
-       }
-    },
-    {
-      name: 'locations',
-      displayKey: 'name',
-      source: locations.ttAdapter(),
-      templates: {
-        header: '<h3 class="dd-category">' + locations_text + '</h3>',
-        suggestion: _.template('<a href="/projects/?location=<%= code %>"><p><%= name %></p></a>')
-       }
-    }
-    // ,
-    // {
-    //   name: 'sectors',
-    //   displayKey: 'name',
-    //   source: sectors.ttAdapter(),
-    //   templates: {
-    //     header: '<h3 class="dd-category">Sectors</h3>',
-    //     suggestion: _.template('<a href="/project/?sector=<%= code %>"><p><%= name %></p></a>')
-    //    }
-    // }
-  );
-
-  $('#filterForm').on('click', '.filter_focus_area', function (event) {
-    var tmp = $('#id_title').val();
-    $('select#id_focus_area option').filter(function() {
-      return $(this).text() == tmp;
-    }).prop('selected', true);
-    $('#id_title').val('');
-    $('#filterForm').submit();
-  });
-
-  $('#filterForm').on('mouseover', '.filter_focus_area', function (event) {
-    $('#collapseOne').addClass('in');
-    $('select#id_focus_area option').filter(function() {
-      return $(this).text() == $(this).val();
-    }).prop('selected', true);
-  });
-
-  $('#filterForm').on('mouseout', '.filter_focus_area', function (event) {
-    $('#collapseOne').removeClass('in');
-  });
-
-  $('#sortByDropdown').on('click', 'a', (function(event) {
-    insertParam('sort_by', $(event.target).attr('param'));
-  }));
+    var locations = new Bloodhound({
+        datumTokenizer: Bloodhound.tokenizers.obj.whitespace('name'),
+        queryTokenizer: Bloodhound.tokenizers.whitespace,
+        prefetch: {
+            //  TODO: get a filtered set of locations for Pages
+            thumbprint: "locations",
+            url: '/static/data/m49.json'
+        }
+    });
+
+    var keywords = new Bloodhound({
+        datumTokenizer: Bloodhound.tokenizers.obj.whitespace('label'),
+        queryTokenizer: Bloodhound.tokenizers.whitespace,
+        prefetch: {
+            url: '/rest/v1/typeaheads/keywords?format=json',
+            thumbprint: "keywords",
+
+            filter: function (response) {
+                return response.results;
+            }
+        }
+    });
+
+    var projects = new Bloodhound({
+        datumTokenizer: Bloodhound.tokenizers.obj.whitespace('title'),
+        queryTokenizer: Bloodhound.tokenizers.whitespace,
+        prefetch: {
+            url: '/rest/v1/typeaheads/projects?format=json&published=1',
+            thumbprint: "projects",
+
+            filter: function (response) {
+                return response.results;
+            }
+        }
+    });
+
+    projects.initialize();
+    locations.initialize();
+    keywords.initialize();
```

Comment:
Yes, I dropped the organisations. But I can easily put them back. One issue with that is that the search results popup become so tall it might drop below the lower edge of the browser 😞  We can mitigate that by only showing 3 items of of each, but that has other issues, obviously...